src/formatDate.js

```diff
@@ -9,6 +43,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const possibleDateParts = ['YYYY', 'YY', 'MM', 'DD'];
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (const datePart of possibleDateParts) {
+    const datePartIndex = fromFormat.indexOf(datePart);
+
+    if (datePartIndex !== -1) {
+      datePartsMatched[datePart] = oldDateParts[datePartIndex];
+    }
+  }
+  datePartsMatched = convertYear(datePartsMatched, fromFormat, toFormat);
```

Comment:
The reassignment of `datePartsMatched` might be confusing to readers. Instead of reassigning, you could return the updated parts directly from the `convertYear` function, which would make the data flow clearer.